lib/fog/to_hash_document.rb

```diff
@@ -5,20 +5,22 @@ def initialize
     end
 
     def characters(string)
-      @value ||= ""
+      @value ||= +''
       @value << string.strip
     end
 
     def end_element(name)
       last = @stack.pop
-      if last.empty? && @value.empty?
-        @stack.last[name.to_sym] = ""
+      @stack.push({}) if @stack.empty?
```

Comment:
Is this needed? I think it worked before, so I'm just trying to understand why this would change. Also, if we do need it, it seems like maybe it should be before last is set, rather than after it is set as a pop on a potentially empty? Maybe I'm just not quite understanding though. Could you explain?

Reply:
Yes this stems from the fact that on line 17, 19 and 21 ( in all branches of the the is statement) there is a  `@stack.last[name.to_sym]`

Imagine you popping the stack on line 13 with `last = @stack.pop`, and it's the last element
then `@stack` would be equal to `[]`
and so `@stack.last` would evaluate to `nil`
and `@stack.last[...]` would raise a `NoMethodError [] for nil`

Reply:
(it was thanks to the unit tests that i was able to detect this)